scipy/interpolate/_bsplines.py

```diff
@@ -33,6 +34,44 @@ def _as_float_array(x, check_finite=False):
         raise ValueError("Array must not contain infs or nans.")
     return x
 
+def _find_left(t, x, k):
+    """
+    Returns the first index of ``t`` at which
+    ``t[i] <= x < t[i + 1]`` == True.
+    If such index can not be found returns the last index that
+    stands for the last element in initial vector of points
+    ``n + k - 2``.
+    """
+    nt = len(t)
+    for i in range(nt - 1):
+        if (t[i] <= x and t[i + 1] > x):
+            return i
+    return nt - k - 2
+
+def _dual_poly(j, k, t, y):
+    """
+    Dual polynomial of the B-spline B_{j,k,t} -
+    polynomial which is associated with B_{j,k,t}:
+    p_{j,k}(y) = (y - t_{j+1})(y - t_{j+2})...(y - t_{j+k})
```

Comment:
Nit: Since this is TeX notation, consider using $ signs to mark that. Since it's a private function, it's OK either way.